pkg/runtime/runtime.go

```diff
@@ -2491,6 +2486,12 @@ func (a *DaprRuntime) processComponentSecrets(component componentsV1alpha1.Compo
 			continue
 		}
 
+		secretStore := a.getSecretStore(secretStoreName)
+		if secretStore == nil {
+			log.Warnf("component %s references a secret store that isn't loaded: %s", component.Name, secretStoreName)
+			return component, secretStoreName
+		}
+
```

Comment:
with this change, the secret store is still considered unready, but the secret is resolved because it is loaded from operator? 
Does the init step complete with a secret store being not ready? 
Also for the case of builtin secret store, if the secret is being resolved by the operator, why are we warning here that the secret store is not loaded?

Reply:
I don't understand, sorry.

The problem I was trying to solve is that if the component references a secret from the built-in K8s secret store, it should be read from the operator and not from a secret store component. This is explained by the existing comment:

```
// If running in Kubernetes, do not fetch secrets from the Kubernetes secret store as they will be populated by the operator.
```

The problem is when the built-in K8s secret store is not initialized. In this case, there's no component loaded caled `"kubernetes"` so `secretStore := a.getSecretStore(secretStoreName)` returns nil. But that's totally fine because that secret store isn't used anyways. So I had to move the check to after the case for using the built-in secret store.

Reply:
this function is being called in `preprocessOneComponent` function lines https://github.com/dapr/dapr/blob/master/pkg/runtime/runtime.go#LL2522-L2526, and when `kubernetes` secret store is not found, it will return that as an unready dependency. 
Does the initialization continue successfully after that, or does dapr runtime wait for this dependency to be ready? 

Also do we want to print the log warning `"component %s references a secret store that isn't loaded: %s" for the `kubernetes` secret store when we know that `--disable-builtin-k9s-secret-store` is given?

Reply:
> Does the initialization continue successfully after that, or does dapr runtime wait for this dependency to be ready?

As long as the component only references secrets from the built-in secret store, initialization will continue successfully and Dapr won't need to wait.

> Also do we want to print the log warning "component %s references a secret store that isn't loaded: %s" for the kubernetessecret store when we know that--disable-builtin-k9s-secret-store` is given?

I would say that's not needed because component secrets are not loaded from the component but through the Dapr Operator. There's an expectation those will "just work" IMHO.